File: Creational/Prototype/prototype.swift
// Implementation

class Person {
    var name: String?
    let surname: String

    init(surname: String) {
        self.surname = surname
    }

    func clone() -> Person {
        return Person(surname: self.surname)
    }
}

// Usage

let prototype = Person(surname: "Petrov")

let ivan = prototype.clone()
ivan.name = "ivan"

let petr = prototype.clone()
petr.name = "petr"

File: Creational/Prototype/Prototype.kt
// Implementation

class Person(var surname: String) {
    lateinit var name: String
    
    fun clone(): Person {
        return Person(surname)
    }
}

// Usage

fun main(args: Array<String>) {
    val prototype = Person(surname = "Petrov")

    val ivan = prototype.clone()
    ivan.name = "Ivan"

    val petr = prototype.clone()
    petr.name = "petr"
}